dask/dataframe/_pyarrow.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd
+
+from dask.dataframe._compat import PANDAS_GT_130, PANDAS_GT_150
+from dask.dataframe.utils import is_dataframe_like, is_index_like, is_series_like
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pa = None
+
+
+PYARROW_STRINGS_AVAILABLE: bool = pa is not None and PANDAS_GT_130
+
+
+def is_pyarrow_string_dtype(dtype):
+    """Is the input dtype a pyarrow string?"""
+    if not PYARROW_STRINGS_AVAILABLE:
+        return False
+
+    if PANDAS_GT_150:
+        types = [pd.StringDtype("pyarrow"), pd.ArrowDtype(pa.string())]
+    else:
+        types = [pd.StringDtype("pyarrow")]
+    if dtype in types:
+        return True
+    return False
+
+
+def is_object_string_dtype(dtype):
+    """Determine if input is a non-pyarrow string dtype"""
+    return pd.api.types.is_string_dtype(dtype) and not is_pyarrow_string_dtype(dtype)
+
+
+def is_object_string_index(x):
+    return (
+        is_index_like(x)
+        and is_object_string_dtype(x.dtype)
+        and not isinstance(
+            x, pd.MultiIndex
+        )  # MultiIndex don't support non-object dtypes
+    )
+
+
+def is_object_string_series(x):
+    return is_series_like(x) and (
+        is_object_string_dtype(x.dtype) or is_object_string_index(x.index)
+    )
+
+
+def is_object_string_dataframe(x):
+    return is_dataframe_like(x) and (
+        any(is_object_string_series(s) for _, s in x.items())
+        or is_object_string_index(x.index)
+    )
+
+
+def to_pyarrow_string(df):
+    if not (is_dataframe_like(df) or is_series_like(df) or is_index_like(df)):
+        return df
+
+    # Possibly convert DataFrame/Series/Index to `string[pyarrow]`
+    dtypes = None
+    if is_dataframe_like(df):
+        dtypes = {
+            col: pd.StringDtype("pyarrow")
+            for col, s in df.items()
```

Comment:
Thanks for the suggestion. I like iterating over `dtypes` regardless because it's a bit more straightforward (don't need to then select the `.dtype` attribute on each `Series`).

Reply:
Yep that's true. It is most certainly faster as well except if you already accessed the same column before, then it is cached and super fast. But this is an edge case here I guess